ref_builder/plan.py

```diff
@@ -99,6 +99,11 @@ def required_segments(self) -> list[SegmentPlan]:
         return [segment for segment in self.segments if segment.required]
 
 
+IsolatePlan = Union[MultipartitePlan, MonopartitePlan]
```

Comment:
* Add docstring.
* I don't think `Isolate` contributes to anything here. We only have one kind of plan at this time.

```suggestion
Plan: TypeAlias = Union[MultipartitePlan, MonopartitePlan]
```

Reply:
This would necessitate `Segment.name` accepting null values again. Should it?